lib/routes/prototype-admin-routes.js

```diff
@@ -421,67 +438,36 @@ router.get('/plugins/install', async (req, res) => {
     currentUrl: req.originalUrl,
     links: managementLinks,
     chosenPlugin,
-    command: req.query.mode === 'upgrade' ? chosenPlugin.upgradeCommand : chosenPlugin.installCommand,
-    verb: {
-      title: req.query.mode === 'upgrade' ? 'Upgrade' : 'Install',
-      para: req.query.mode === 'upgrade' ? 'upgrade' : 'install',
-      status: req.query.mode === 'upgrade' ? 'upgraded' : 'installed'
-    }
+    command: getCommand(req.params.mode, chosenPlugin),
+    verb: getVerb(req.params.mode)
   })
 })
 
-router.post('/plugins/install', async (req, res) => {
-  const chosenPlugin = await getPluginForRequest(req)
-  res.render(getManagementView('plugin-install-process.njk'), {
-    chosenPlugin,
-    verb: {
-      title: req.query.mode === 'upgrade' ? 'Upgrading' : 'Installing',
-      para: req.query.mode === 'upgrade' ? 'upgrading' : 'installing'
-    }
-  })
-  try {
-    await exec(chosenPlugin.installCommand, { cwd: projectDir })
-  } catch (e) {
-    res.status(500).send(['Something went wrong', e.message, e.stack].join('br/'))
-  }
-})
+const startTimestamp = Date.now()
 
-router.get('/plugins/uninstall', async (req, res) => {
-  const pageName = 'Plugins'
-  const chosenPlugin = await getPluginForRequest(req)
-
-  if (!chosenPlugin) {
-    res.status(404).send('Plugin not found')
-    return
-  }
-  res.render(getManagementView('plugin-install-or-uninstall.njk'), {
-    currentPage: pageName,
-    currentUrl: req.originalUrl,
-    links: managementLinks,
-    command: chosenPlugin.uninstallCommand,
-    chosenPlugin,
-    verb: {
-      title: 'Uninstall',
-      para: 'uninstall',
-      status: 'uninstalled'
-    }
-  })
+router.post('/plugins/status', async (req, res) => {
+  res.json({ startTimestamp })
 })
 
-router.post('/plugins/uninstall', async (req, res) => {
-  const chosenPlugin = await getPluginForRequest(req)
-  res.render(getManagementView('plugin-install-process.njk'), {
-    chosenPlugin,
-    verb: {
-      title: 'Uninstalling',
-      para: 'uninstalling'
-    }
-  })
+router.post('/plugins/:mode', async (req, res) => {
   try {
-    await exec(chosenPlugin.uninstallCommand, { cwd: projectDir })
+    const chosenPlugin = await getPluginForRequest(req)
+    const command = getCommand(req.params.mode, chosenPlugin)
+    exec(command, { cwd: projectDir }).finally(() => {
+      console.log(`Completed ${command}`)
+      // force the application to stop after a delay as nodemon restart does not always work on Windows when running acceptance tests
+      if (config.isAcceptanceTest) {
+        setTimeout(() => {
+          process.exit(1)
+        }, 2000)
+      }
+    })
   } catch (e) {
-    res.status(500).send(['Something went wrong', e.message, e.stack].join('br/'))
+    console.log(e)
   }
+  res.json({ startTimestamp })
 })
 
+router.use('/scripts', express.static(path.join(assetsDir, 'javascripts', 'manage-prototype')))
```

Comment:
I suggest we use the Plugins framework for this - you can put it in `assets`. I can help with that if you want.

Reply:
Done, thanks for your help @nataliecarey